src/apps/ENGINE/compiler.cpp

```diff
@@ -1088,9 +1088,11 @@ void COMPILER::ProcessFrame(uint32_t DeltaTime)
 
 void COMPILER::ResizeBCodeBuffer(SEGMENT_DESC &Segment, uint32_t add_size)
 {
-    if ((Segment.BCode_Program_size + add_size) >= Segment.BCode_Buffer_size)
+    int64_t needed_memory = static_cast<int64_t>(Segment.BCode_Program_size) + add_size - Segment.BCode_Buffer_size;
+    if (needed_memory >= 0)
```

Comment:
if needed_memory == 0, then needed_memory - 1 == -1, (needed_memory - 1) / BCODE_BUFFER_BLOCKSIZE == 0 (in Python it would be -1, yes) and (needed_memory - 1) / BCODE_BUFFER_BLOCKSIZE + 1 == 1: 1 block will be allocated. The same thing was done in the old code, a block would be allocated even if the difference was 0.